libmpv/render.h

```diff
@@ -164,6 +164,18 @@ typedef enum mpv_render_param_type {
      * Type: struct wl_display*
      */
     MPV_RENDER_PARAM_WL_DISPLAY = 9,
+    /**
+     * DRM display, contains drm display handles.
+     * Valid for mpv_render_context_create().
+     * Type : struct mpv_opengl_drm_params*
+     */
+    MPV_RENDER_PARAM_DRM_DISPLAY = 10,
+    /**
+     * DRM osd size, contains osd dimensions.
+     * Valid for mpv_render_context_create().
+     * Type : struct mpv_opengl_drm_osd_size*
+     */
+    MPV_RENDER_PARAM_DRM_OSD_SIZE = 11,
```

Comment:
Apparently your code expects that the pointers the user passes are valid for the life time of the context, which I'd find surprising. I think the structs really should be copied (and then use pointers for fields which somehow need to be changed per frame or something). That would require changes in libmpv_gpu.c (init() function). I guess native_resource_map needs to be changed into an array of structs, so a struct size to copy can be specified.

Reply:
The drmModeEtomicRequest pointer which is what you're referring to as changing every frame is in that struct and already passed as a pointer to the pointer address. So copying this should still be possible.
But yes you are right on the fact that I have been assuming that the struct keeps being alive during the whole playback. If some change is made to the native_resource_map, we can probably use a copy to overcome this.

Reply:
Added such a change here: https://github.com/mpv-player/mpv/pull/5716/commits/0bfdb0029dd38350169677a38601a17f8c1a727a
You'd simply set the .size field to the struct sizes, and it'd copy them.

Reply:
That's part of #5716 btw., which somehow turned into a PR for general changes.